internal/oci/notation/notation.go

```diff
@@ -0,0 +1,388 @@
+// notation public keys.
+const DefaultTrustPolicyKey = "trustpolicy.json"
+
+// options is a struct that holds options for verifier.
+type options struct {
+	rootCertificate [][]byte
+	rOpt            []remote.Option
+	trustPolicy     *trustpolicy.Document
+	auth            authn.Authenticator
+	keychain        authn.Keychain
+	insecure        bool
+	logger          logr.Logger
+}
+
+// Options is a function that configures the options applied to a Verifier.
+type Options func(opts *options)
+
+// WithInsecureRegistry sets notation to verify against insecure registry.
+func WithInsecureRegistry(insecure bool) Options {
+	return func(opts *options) {
+		opts.insecure = insecure
+	}
+}
+
+// WithTrustStore sets the trust store configuration.
+func WithTrustStore(trustStore *trustpolicy.Document) Options {
+	return func(opts *options) {
+		opts.trustPolicy = trustStore
+	}
+}
+
+// WithRootCertificate is a functional option for overriding the default
+// rootCertificate options used by the verifier to set the root CA certificate for notary.
+// It takes in a list of certificate data as an array of byte slices.
+// The function returns a options function option that sets the public certificate
+// in the notation options.
+func WithRootCertificate(data [][]byte) Options {
+	return func(opts *options) {
+		opts.rootCertificate = data
+	}
+}
+
+// WithRemoteOptions is a functional option for overriding the default
+// remote options used by the verifier
+func WithRemoteOptions(opts ...remote.Option) Options {
+	return func(o *options) {
+		o.rOpt = opts
+	}
+}
+
+// WithAuth is a functional option for overriding the default
+// authenticator options used by the verifier
+func WithAuth(auth authn.Authenticator) Options {
+	return func(o *options) {
+		o.auth = auth
+	}
+}
+
+// WithKeychain is a functional option for overriding the default
+// keychain options used by the verifier
+func WithKeychain(key authn.Keychain) Options {
+	return func(o *options) {
+		o.keychain = key
+	}
+}
+
+// WithLogger is a function that returns an Options function to set the logger for the options.
+// The logger is used for logging purposes within the options.
+func WithLogger(logger logr.Logger) Options {
+	return func(o *options) {
+		o.logger = logger
+	}
+}
+
+// NotationVerifier is a struct which is responsible for executing verification logic
+type NotationVerifier struct {
+	auth     authn.Authenticator
+	keychain authn.Keychain
+	verifier *notation.Verifier
+	opts     []remote.Option
+	insecure bool
+	logger   logr.Logger
+}
+
+var _ truststore.X509TrustStore = &trustStore{}
+
+// trustStore is used by notation-go/verifier to retrieve the root certificate for notary.
+// The default behaviour is to read the certificate from disk and return it as a byte slice.
+// The reason for implementing the interface here is to avoid reading the certificate from disk
+// as the certificate is already available in memory.
+type trustStore struct {
+	cert [][]byte
```

Comment:
```suggestion
	certs [][]byte
```

Reply:
Resolved in [271aaa4](https://github.com/fluxcd/source-controller/pull/1075/commits/271aaa4961eed43aef8ca7d9f54f252047fcd340).